src/Compiler/Service/ServiceDeclarationLists.fs

```diff
@@ -1138,6 +1138,11 @@ type DeclarationListInfo(declarations: DeclarationListItem[], isForType: bool, i
             // Remove all duplicates. We've put the types first, so this removes the DelegateCtor and DefaultStructCtor's.
             |> RemoveDuplicateCompletionItems g
             |> List.groupBy (fun x ->
+                // RFC-1137
```

Comment:
Just for my understanding of the code flow - how does this information propagate throught the code?

Reply:
Before, it used to `groupBy` on the name only (to discard results, as only `items.Head` in account in the subsequent stage of the pipeline).

Now it `groupBy` on a tuple with `fst` being value which is either `None` or `Some CompletionItemKind.Method` and `snd` being the name of the item.

Effectively, it singles out the extension by name to always have them in the list, while before, they were hidden by properties / members of the same name.

It is actually not a correct change, as 

```fsharp
type WithEvent()= 
  let ev = Event<string>()
  [<CLIEvent>]
  member x.Ev = ev.Publish

module Exts =
  type WithEvent with
    member x.Ev v = x

open Exts

let we = WithEvent()
we.Ev // it shows both here, while it is impossible to call anything else than the event
```

I don't think we can ship with this issue, and possibly the change in the logic should only apply if the language feature is present.

The main issue is, the whole pipeline is complex and may require overhaul to achieve what is needed, I need to ponder on a better solution.

Reply:
Anyone knows if there are tests around `DeclarationListInfo.Create`? It seems like super sensible and complex area of FCS, I'd like to add non regression test for the above case, and probably others; but it is a bit complex to do it if we don't have the infrastructure for it already.